### PR TITLE
Update utils.py

### PR DESCRIPTION
**Pull request by nmegiddo**
Use a different algorithm for  norm=1, based on exponential distribution.

# Description

A sample point is generated by a L1-normalized vector (in dimenion nb_dims + 1) of IID unit-exponential variables, then 
dropping the additional dimension and flipping signs randomly.

Fixes # (issue)

## Type of change

Please check all relevant options.

- [v] Improvement (non-breaking)
- [v] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [ ] Test A
- [ ] Test B

**Test Configuration**:
- OS
- Python version
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by beat-buesser — APPROVED** [review]
Hi @nmegiddo Thank you very much for accelerating the sampling in norm 1!
